### PR TITLE
Add 3 cases of virt-v2v: multi-kernel

### PR DESCRIPTION
**Pull request by chloerh**
Add 3 cases of os-specific: multiple kernel

### CODE REVIEW

File: v2v/tests/src/v2v_options.py [hunk 1]
```diff
@@ -53,12 +55,18 @@ def run(test, params, env):
     v2v_user = params.get("unprivileged_user", "")
     v2v_timeout = int(params.get("v2v_timeout", 1200))
     status_error = "yes" == params.get("status_error", "no")
-    for param in [vm_name, remote_host, esx_ip, vpx_dc, ovirt_engine_url,
+    vm_user = params.get("vm_user")
+    vm_pwd = params.get("vm_pwd")
+    checkpoint = params.get("checkpoint")
+    debug_kernel = 'debug_kernel' == params.get('checkpoint')
+    os_type = params.get("os_type", "linux")
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: So "debug_kernel = 'yes'" in the config file is usless.
And seems the os_type is never used in this file

File: v2v/tests/src/v2v_options.py [hunk 2]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+                               env.get("address_cache"))
+            session = vm_login(vm)
+            kwargs["session"] = session
+            func(*args, **kwargs)
+            if session:
+                session.close()
+            vm.shutdown()
+        return wrapper
+
+    def vm_login(vm):
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: There's ```wait_for_login``` func in libvirt_vm,  you don't need to redefine it.

File: v2v/tests/src/v2v_options.py [hunk 3]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+                logging.info('Retrying...')
+                retry -= 1
+                time.sleep(10)
+        raise error.TestError('Connect to VM failed')
+
+    def install_kernel(session, url=None, debug=False):
+        """
+        Install kernel to vm
+        """
+        if debug:
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: Here you mean to install debug kernel, so the keywork ```debug``` here is not appropriate, may confused with code/case debug level.

File: v2v/tests/src/v2v_options.py [hunk 4]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+        # Check if kernel already exists
+        kernels = session.cmd('rpm -q kernel')
+        logging.debug(kernels)
+        cmd_get_kernel = "wget %s" % url
+        kernel_fname = url.split('/')[-1]
+        kernel_name = '.'.join(kernel_fname.split('.')[:-1])
+        logging.debug('kernel to install is %s' % kernel_name)
+        if kernel_name in kernels:
+            raise error.TestError('Kernel %s already exists!' % kernel_name)
+        cmd_install = "rpm -iv %s" % kernel_fname
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: Using 'yum install $URL', so you don't need to download it by wget

File: v2v/tests/src/v2v_options.py [hunk 5]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+            else:
+                logging.info('install debug kernel success')
+            return 'kernel-debug'
+
+        # Check if kernel already exists
+        kernels = session.cmd('rpm -q kernel')
+        logging.debug(kernels)
+        cmd_get_kernel = "wget %s" % url
+        kernel_fname = url.split('/')[-1]
+        kernel_name = '.'.join(kernel_fname.split('.')[:-1])
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: As the url is given by user, how could you make sure the format it correct(match) like your assumption.

File: v2v/tests/src/v2v_options.py [hunk 6]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+            fm_name = kernel_fm_url.split('/')[-1]
+            cmd_get_kernel_fm = 'wget %s' % kernel_fm_url
+            cmd_install_kernel_fm = 'rpm -Uv %s' % fm_name
+            fm = '.'.join(fm_name.split('.')[:-1])
+            exist_fm = session.cmd("rpm -q kernel-firmware",
+                                   ignore_all_errors=True)
+            if exist_fm is None or fm not in exist_fm:
+                session.cmd(cmd_get_kernel_fm, timeout=v2v_timeout)
+                session.cmd(cmd_install_kernel_fm, timeout=v2v_timeout)
+                session.cmd("rm -f %s" % fm_name)
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: Same with above.
A digression, if you want create/define any tmp file/dir, please put it under ```data_dir.get_tmp_dir()```

File: v2v/tests/src/v2v_options.py [hunk 7]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+        session.cmd("rm -f %s" % kernel_fname)
+        return kernel_name
+
+    @vm_shell
+    def cleanup_kernel(kernel, debug=False, **kwargs):
+        """
+        Clean up test kernel and revert grub entry
+        """
+        session = kwargs["session"]
+        if debug:
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: Same problem

File: v2v/tests/src/v2v_options.py [hunk 8]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+            kernel_to_clean = 'kernel-debug'
+        else:
+            kernel_to_clean = kernel
+        logging.info('Removing test kernel')
+        session.cmd('rpm -e %s' % kernel_to_clean)
+        if session.cmd_status("ls /boot/grub/grub.conf"):
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: This file should be for old grub, not grub2

File: v2v/tests/src/v2v_options.py [hunk 9]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+            session.cmd("grub2-set-default 0")
+        else:
+            session.cmd("sed -i 's/default=./default=0/' /boot/grub/grub.conf")
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: So here you restore the default kernel to 0, but if the original one it not 0, should we consider this?

File: v2v/tests/src/v2v_options.py [hunk 10]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+        """
+        if debug:
+            logging.info('installing kernel-debug')
+            params['installed_kernel'] = install_kernel(session, debug=True)
+        elif input_mode == 'libvirt' and hypervisor == 'kvm':
+            logging.info('For kvm: Installing kernel...')
+            kernel1_url = params.get('kernel1_url')
+            params['installed_kernel'] = install_kernel(session, kernel1_url)
+        else:
+            raise error.TestNAError('Not for multi-kernel')
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: Deal with the input_mode outside

File: v2v/tests/src/v2v_options.py [hunk 11]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+
+        if not session.cmd_status('cat /boot/grub/menu.lst'):
+            grub = 1
+            out = session.cmd(
+                    "cat /boot/grub/menu.lst |grep 'initrd /initramfs-'"
+            )
+            kernel_lst = re.findall('initrd /initramfs-.*', out)
+            logging.debug(kernel_lst)
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: About all these debug/info/error messages
1. Put them in appropriate places
2. The message should be clear and consie

File: v2v/tests/src/v2v_options.py [hunk 12]
```diff
@@ -270,6 +279,203 @@ def check_disks(ori_disks):
...
+            session.cmd(cmd_set_grub)
+            kernel_to_set = kernel_lst[seq]
+        elif grub == 2:
+            kernel_to_set = kernel_lst[seq].split("'")[1].strip("'")
+            cmd_set_grub = 'grub2-set-default "%s"' % kernel_to_set
+            session.cmd(cmd_set_grub)
+        else:
+            raise error.TestFail("Grub version Unknown")
+
+        return kernel_to_set
```
Comment (will-Do) [inline v2v/tests/src/v2v_options.py]: Please check virttest/libvirt_vm.py: set_kernel_param()
You can split that function to 1. get_grub_file() 2. set_kernel_param(), then add a new function like: set_boot_kernel()

**Comment by chloerh** [comment]
Modified code as instructions. Please review. Thanks
test results:

for rhel7.1
$ sudo avocado run --vt-type v2v type_specific.local.v2v_options.positive_test.specific_vm.input_mode.libvirt.kvm.multiple_kernel
JOB ID     : 8e2855b85d580dca9c72ee82d862c24aec49c828
JOB LOG    : /root/avocado/job-results/job-2016-03-11T12.11-8e2855b/job.log
TESTS      : 3
 (1/3) type_specific.local.v2v_options.positive_test.specific_vm.input_mode.libvirt.kvm.multiple_kernel.default.output_mode.rhev: FAIL (608.37 s)
 (2/3) type_specific.local.v2v_options.positive_test.specific_vm.input_mode.libvirt.kvm.multiple_kernel.vmlinuz_init.output_mode.rhev: PASS (658.66 s)
 (3/3) type_specific.local.v2v_options.positive_test.specific_vm.input_mode.libvirt.kvm.multiple_kernel.debug_kernel.output_mode.rhev: PASS (722.92 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0
JOB HTML   : /root/avocado/job-results/job-2016-03-11T12.11-8e2855b/html/results.html
TIME       : 1989.94 s

for rhel6.6
$ sudo avocado run --vt-type v2v type_specific.local.v2v_options.positive_test.specific_vm.input_mode.libvirt.kvm.multiple_kernel
JOB ID     : 52e2690ecd85df962d637032e2e23d9042b7cff9
JOB LOG    : /root/avocado/job-results/job-2016-03-11T12.46-52e2690/job.log
TESTS      : 3
 (1/3) type_specific.local.v2v_options.positive_test.specific_vm.input_mode.libvirt.kvm.multiple_kernel.default.output_mode.rhev: FAIL (789.01 s)
 (2/3) type_specific.local.v2v_options.positive_test.specific_vm.input_mode.libvirt.kvm.multiple_kernel.vmlinuz_init.output_mode.rhev: PASS (807.67 s)
 (3/3) type_specific.local.v2v_options.positive_test.specific_vm.input_mode.libvirt.kvm.multiple_kernel.debug_kernel.output_mode.rhev: FAIL (807.10 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0
JOB HTML   : /root/avocado/job-results/job-2016-03-11T12.46-52e2690/html/results.html
TIME       : 2403.78 s

**Comment by chloerh** [comment]
for failed cases, 2 existed bug

virt-v2v always use the first kernel (0th kernel) instead of the default kernel (ex: default=1)
https://bugzilla.redhat.com/show_bug.cgi?id=829859

virt-v2v picks debug kernels over non-debug kernels when versions are equal https://bugzilla.redhat.com/show_bug.cgi?id=1170073

**Pull request closed without merging**